internal/providers/declaration_info.bzl

```diff
@@ -41,3 +32,21 @@ This prevents needing an aspect in rules that consume the typings, which improve
         "type_blacklisted_declarations": """A depset of .d.ts files that we should not use to infer JSCompiler types (via tsickle)""",
     },
 )
+
+def declaration_info(declarations, deps = []):
+    """Constructs a DeclarationInfo including all transitive declarations from DeclarationInfo providers in a list of deps.
+
+Returns a single DeclarationInfo.
+"""
+    transitive_depsets = [declarations]
+    for dep in deps:
```

Comment:
keep the TODO here - we want to create an action to validate the .d.ts files at this point so a downstream compilation doesn't fail spooky-action-at-a-distance (and maybe use `--skipLibCheck` more safely)